lib/src/features/home/story/view/add_story_view.dart

```diff
@@ -104,21 +124,24 @@ class AddEventBar extends StatelessWidget {
           IconButtonAddEvent(
             icon: const Icon(Icons.camera_alt),
             handlePress: () {
-              context.read<AddStoryBloc>().pickImage(ImageSource.camera);
+              addStoryBloc.pickImage(ImageSource.camera);
             },
           ),
           const SizedBox(height: 20),
           IconButtonAddEvent(
             icon: const Icon(Icons.image),
             handlePress: () {
-              context.read<AddStoryBloc>().pickImage(ImageSource.gallery);
+              addStoryBloc.pickImage(ImageSource.gallery);
             },
           ),
           const SizedBox(height: 20),
           IconButtonAddEvent(
             icon: const Icon(Icons.event_available),
-            handlePress: () {
-              showMaterialModalBottomSheet(
+            handlePress: () async {
+              await addStoryBloc.getEvents();
+
+              // ignore: use_build_context_synchronously
+              final result = await showMaterialModalBottomSheet<MEvent>(
```

Comment:
Create class XBottomSheet and add method showCustomBottomSheet to reuse (Similar to previous PR), the custom UI will be passed into that method